src/runtime/pythonengine.cs

```diff
@@ -303,6 +311,8 @@ public static void Shutdown()
                 _pythonPath = IntPtr.Zero;
 
                 Runtime.Shutdown();
+
+                AppDomain.CurrentDomain.DomainUnload -= OnDomainUnload;
```

Comment:
I remember I got crashes on my side when trying to remove the callback like that. I verified that the list of callbacks was fresh on init (did not contain OnDomainUnload).
Maybe something else was involved in the crash

Reply:
I thought I'd just cherry-picked this part, perhaps I missed a commit?

Anyway, no crash at the moment, we can get more complicated if we need to in a future commit.